artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -1323,7 +1328,13 @@ private PagedReference moveNext() {
             PagePositionAndFileOffset lastPosition = position;
             PagePositionAndFileOffset tmpPosition = position;
 
+            long timeout = System.currentTimeMillis() + DELIVERY_TIMEOUT;
```

Comment:
Use nanoTime here that is not susceptible to NTP updates :)
Now I can return to my honeymoon :P

Reply:
Resolved it, thanks :)